### PR TITLE
add intermediate e2e runs to GCE upgrade suite to mirror GKE upgrade suite

### PR DESCRIPTION
**Pull request by mikedanese**
@kubernetes/goog-testing

### CODE REVIEW

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -612,26 +612,76 @@ case ${JOB_NAME} in
...
+    ;;
+
+  kubernetes-upgrade-gce-step5-e2e-old)
+    : ${E2E_CLUSTER_NAME:="gce-upgrade"}
+    : ${E2E_NETWORK:="gce-upgrade"}
+    : ${E2E_OPT:="--check_version_skew=false"}
+    : ${JENKINS_FORCE_GET_TARS:=y}
+    : ${JENKINS_USE_RELEASE_TARS:=y}
+    : ${PROJECT:="kubernetes-jenkins-gke-upgrade"}
+    : ${PROJECT:="k8s-jkns-gce-upgrade"}    : ${E2E_UP:="false"}
```
Comment (ihmccreery) [inline hack/jenkins/e2e.sh]: Typo: should be on separate lines, and there should only be one `PROJECT`

File: hack/jenkins/e2e.sh [hunk 2]
```diff
@@ -612,26 +612,76 @@ case ${JOB_NAME} in
     : ${E2E_UP:="false"}
     : ${E2E_TEST:="true"}
     : ${E2E_DOWN:="false"}
-    : ${GINKGO_TEST_ARGS:="--ginkgo.focus=Skipped.*Cluster\supgrade.*upgrade-cluster"}
-    NUM_MINIONS=3
+    : ${GINKGO_TEST_ARGS:="--ginkgo.focus=Skipped.*Cluster\supgrade.*upgrade-master"}
+    : ${NUM_MINIONS:=3}
```
Comment (ihmccreery) [inline hack/jenkins/e2e.sh]: Do you know why we're running these on clusters of 3, as opposed to just using the default, (not sure what that is—2?)?

Comment (mikedanese) [inline hack/jenkins/e2e.sh]: default is 3 https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/config-test.sh#L23. I'm being explicit here to work around some weird behavior I was seeing where some steps were not picking up the test config. I admit that I have not fully investigated why but the reason was not obvious.

Comment (ihmccreery) [inline hack/jenkins/e2e.sh]: Okay.  Can you file an issue (if there isn't already one) and add TODO(<issue#>)s?

**Comment by ihmccreery** [comment]
LGTM.  Sorry about the delay, @mikedanese.  Next time, let me know when you want me to take a look so I get an email?

**Comment by ihmccreery** [comment]
Sorry @mikedanese; I've been changing things in the way we version test targets to allow us to do this a bit more sanely (Use JENKINS_PUBLISHED_VERSION instead of JENKINS_USE_RELEASE_TARS and JENKINS_USE_STABLE #15168).

**Pull request merged**